sw/host/opentitantool/src/command/spi.rs

```diff
@@ -0,0 +1,276 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use erased_serde::Serialize;
+use indicatif::{ProgressBar, ProgressStyle};
+use std::fs::{self, File};
+use std::io::{self, Write};
+use std::time::Instant;
+use structopt::StructOpt;
+
+use opentitanlib::app::command::CommandDispatch;
+use opentitanlib::io::spi::Transfer;
+use opentitanlib::spiflash::SpiFlash;
+use opentitanlib::transport::{Capability, Transport};
+
+/// Read and parse an SFDP table.
+#[derive(Debug, StructOpt)]
+pub struct SpiSfdp {
+    #[structopt(
+        short,
+        long,
+        help = "Display raw SFDP bytes rather than the parsed struct."
+    )]
+    raw: Option<usize>,
+}
+
+// Print a hexdump of a buffer to stdout.
+// The hexdump includes the offset, hex bytes and printable ASCII characters.
+//
+//  00000000: 53 46 44 50 06 01 02 ff 00 06 01 10 30 00 00 ff  SFDP........0...
+//  00000010: c2 00 01 04 10 01 00 ff 84 00 01 02 c0 00 00 ff  ................
+//  00000020: ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff  ................
+//  00000030: e5 20 fb ff ff ff ff 3f 44 eb 08 6b 08 3b 04 bb  . .....?D..k.;..
+//
+// Note: This format can be consumed by `xxd -r` and converted back into binary.
+fn hexdump(buf: &[u8]) {
+    let mut ascii = [b'.'; 16];
+    for (i, v) in buf.iter().enumerate() {
+        if i % 16 == 0 {
+            if i != 0 {
+                // Print the ASCII decode.
+                println!("  {}", String::from_utf8_lossy(&ascii));
+            }
+            // Print the offset.
+            print!("{:08x}:", i);
+        }
+        // Print each byte.
+        print!(" {:02x}", *v);
+        // For printable ASCII chars, place them in the ascii buffer.
+        // Replace all non-printable chars with a dot.
+        ascii[i % 16] = if *v >= 32 && *v < 127 { *v } else { b'.' };
```

Comment:
You can make the conditon be `*v == b' ' || v.is_ascii_graphic()`.

Reply:
Sigh.  I know its "more correct" to use the helper functions, but the range of printable ASCII characters is fixed forever.  I suppose its more a statement about the timeframe in which I learned to code that I just happen to have memorized certain features of the ASCII character set.